locking/locks.go

```diff
@@ -92,7 +93,8 @@ func (c *Client) LockFile(path string) (Lock, error) {
 		return Lock{}, fmt.Errorf("Error caching lock information: %v", err)
 	}
 
-	return lock, nil
+	// Ensure writeable on return
+	return lock, tools.SetFileWriteFlag(path, true)
```

Comment:
I dislike the pattern of returning a single error via a method call in the return signature, because I think this hides too much. I prefer:

```go
if err := tools.SetFileWriteFlag(path, true); err != nil {
        return Lock{}, err
}

return lock, nil